File: Data Structures & Algorithms/anagram-groups/submission-3.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        HashMap<String, List<String>> seen = new HashMap();

        for(String str : strs){
            int[] count = new int[26];
            
            for(char chr : str.toCharArray()){
                count[chr - 'a']++;
            }

            String key = Arrays.toString(count);
            seen.putIfAbsent(key, new ArrayList<>());
            seen.get(key).add(str);
        }

        return new ArrayList<>(seen.values());   
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        i = 0

        while nums[i] not in seen:
            seen[(target - nums[i])] = i
            i += 1

        return [seen[nums[i]], i]